### PR TITLE
add a rest client config for configuring polling and retries on ARM clients

### PR DESCRIPTION
**Pull request by marwanad**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

adds a rest client config for configuring polling and retries on ARM clients. Consumers creating AZ clients can specify different configurations.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
N/A
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by marwanad** [comment]
@andyzhangx @feiskyer tried to keep the interface change minimal but not sure if that would possibly break other consumers (apart from autoscaler). Do disk drivers vendor anything from here?

File: pkg/azureclients/armclient/azure_armclient.go [hunk 1]
```diff
@@ -75,20 +76,35 @@ type Client struct {
 }
 
 // New creates a ARM client
-func New(authorizer autorest.Authorizer, baseURI, userAgent, apiVersion, clientRegion string, clientBackoff *retry.Backoff) *Client {
-	restClient := autorest.NewClientWithUserAgent(userAgent)
-	restClient.PollingDelay = 5 * time.Second
-	restClient.RetryAttempts = 3
-	restClient.RetryDuration = time.Second * 1
+func New(authorizer autorest.Authorizer, clientConfig azureclients.ClientConfig, baseURI, apiVersion string) *Client {
```
Comment (marwanad) [inline pkg/azureclients/armclient/azure_armclient.go:79]: using the client configs reduces the number of parameter but it feels odd to have it as a dependency from `azureclients`

**Comment by feiskyer** [comment]
OK, capz tests failed again

/retest

**Comment by jackfrancis** [comment]
@CecileRobertMichon let’s prioritize investigating this test failure today

**Comment by jackfrancis** [comment]
@feiskyer the previous capz failure was caused by a github artifacts outage, see:

https://github.com/kubernetes-sigs/cloud-provider-azure/pull/925#issuecomment-988812621

I'm referring to this test failure:

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cloud-provider-azure/925/pull-cloud-provider-azure-e2e-ccm-capz/1468569315451080704/build-log.txt

Thanks for re-testing, looks like the capz test is green for this PR.

**Comment by feiskyer** [comment]
@jackfrancis thanks for the investigation.

**Pull request merged**